src/core/ucc_coll.c

```diff
@@ -258,7 +311,13 @@ static ucc_status_t ucc_triggered_coll_complete(ucc_coll_task_t *parent_task, //
 {
     ucc_trace("triggered collective complete, task %p, seq_num %u",
               task, task->seq_num);
-    return ucc_ec_task_end(task->ee_task, task->ee->ee_type);
+    if (!(task->flags & UCC_COLL_TASK_FLAG_EXECUTOR)) {
```

Comment:
i would add an explanation comment here:
for triggered post we always have executor even if task does not use one. In that case we must finalize executor here explicitely.

Reply:
This is a special. When i said "task does not use one" i meant that task is not going to post reduce/memcpy operations to exectutor. However, if it is a triggered operation, the executor is still required (since it implements dependency/wait kernel). This is why it will here.